src/mixins/navigator/tagsProvider.js

```diff
@@ -47,17 +48,23 @@ export default {
   },
   methods: {
     extractTags(nodes, apiChanges) {
-      const possibleTags = new Set(Object.values(FILTER_TAGS));
+      const possibleTags = new Set(Object.values(TYPE_TAGS));
       // create categories to order the availableTags
       const availableTags = {
         type: [],
         changes: [],
         other: [],
       };
 
-      // when API changes are available, remove the hide deprecated option
-      if (apiChanges && Object.values(apiChanges).length) {
-        possibleTags.delete(FILTER_TAGS.hideDeprecated);
+      const availableApiChanges = new Set(Object.values(apiChanges));
+      if (availableApiChanges.size) {
+        // only add change tag option if it's present
+        availableApiChanges.forEach((changeType) => {
```

Comment:
What is the value of going through each available API Changes and add them to the `possibleTags` ?

`availableTags` is the returning Array with all the tags. We only use `possibleTags` as a reference to delete the tags that have been selected already to end early in case we already added them to `availableTags`.

We check `FILTER_TAGS[changeType]` later on to push the tag into the `availableTags` already.

Reply:
If API Changes is not on, then we can end early by removing the API Changes tag from the pool of possible tags. 

In the previous implementation, if the API Changes is not on, these tags are still in the list of possible tags. So we have to iterate to the end of the nodes to determine that none of the API Changes tags are available to be added to `availableTags`.

Reply:
Got it, sounds good, thanks for the explanation!